ui/apps/platform/src/Containers/Vulnerabilities/utils/telemetry.ts

```diff
@@ -0,0 +1,25 @@
+import { OnSearchPayload } from 'Components/CompoundSearchFilter/types';
+import {
+    AnalyticsEvent,
+    WORKLOAD_CVE_FILTER_APPLIED,
+    isSearchCategoryWithFilter,
+} from 'hooks/useAnalytics';
+
+type FilterAppliedEvent = typeof WORKLOAD_CVE_FILTER_APPLIED;
+
+export function createFilterTracker(analyticsTrack: (analyticsEvent: AnalyticsEvent) => void) {
+    return function trackAppliedFilter(event: FilterAppliedEvent, payload: OnSearchPayload) {
+        const { action, category, value: filter } = payload;
+
+        if (action !== 'ADD') {
+            // Only track when a filter is applied, not removed
+            return;
+        }
+
+        const telemetryEvent = isSearchCategoryWithFilter(category)
+            ? { event, properties: { category, filter } }
+            : { event, properties: { category } };
+
+        analyticsTrack(telemetryEvent);
+    };
+}
```

Comment:
This minimizes the amount of code we need to write in each place the CompoundSearchFilter is used, but we might still want to evaluate moving tracking inside that component in the future if we think it will be beneficial.